### PR TITLE
fix: Explore Page: Event card alignment

### LINKED ISSUE #7156: Improper alignment of event card
**Describe the bug**

The event card is having improper alignment as the size of the image is larger than the card. 

**To Reproduce**
1. Go to 'https://eventyay.com/explore?category=Science%26 Technology'
2. Scroll down to 'events list'
3. See an error

**Expected behavior**

The image of the card should be the same size as that of the card.

**Screenshots**

image

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser: Firefox

### PR DESCRIPTION
**Pull request by sans2801**
Fixes #7156

#### Short description of what this resolves: Initially the event card on the event search page was having improper alignment as the size of the image was larger than that of the card.

image

Which has been fixed to

image

#### Changes proposed in this pull request: 
- Changing the height property of the div card element from '100%' to 'auto' if the element is wide else '100%'
- Rectified the issues in the previous PR made #7223 

#### Checklist
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by MeghalBisht — DISMISSED** [review]

**Review by rajatgupta24 — APPROVED** [review]

File: app/templates/components/event-card.hbs [hunk 1]
```diff
@@ -9,7 +9,7 @@
       </div>
     {{/unless}}
   {{/if}}
-  <div style="height: 100%;"
+  <div style="{{if this.isWide 'height: auto;' 'height: 100%;'}}"
```
Comment (MeghalBisht) [inline app/templates/components/event-card.hbs:12]: use class instead of style, create classes in utils.scss `h-auto` & `h-100 `respectively

Comment (sans2801) [reply app/templates/components/event-card.hbs:12]: @MeghalBisht yes ma'am

File: app/styles/partials/utils.scss [hunk 2]
```diff
@@ -18,6 +18,14 @@
   display: none !important;
 }
 
+.height.h-auto {
```
Comment (MeghalBisht) [inline app/styles/partials/utils.scss:21]:
```suggestion
.h-auto {
```

File: app/styles/partials/utils.scss [hunk 3]
```diff
@@ -18,6 +18,14 @@
...
+  height: auto;
+}
+
+.height.h-100 {
```
Comment (MeghalBisht) [inline app/styles/partials/utils.scss:25]:
```suggestion
.h-100 {
```

File: app/templates/components/event-card.hbs [hunk 4]
```diff
@@ -9,8 +9,7 @@
       </div>
     {{/unless}}
   {{/if}}
-  <div style="height: 100%;"
-    class="ui card {{this.eventClass}}">
+  <div class="ui card {{this.eventClass}} {{if this.isWide 'height h-auto' 'height h-100'}}">
```
Comment (MeghalBisht) [inline app/templates/components/event-card.hbs:12]:
```suggestion
  <div class="ui card {{this.eventClass}} {{if this.isWide 'h-auto' 'h-100'}}">
```

File: app/styles/partials/utils.scss [hunk 5]
```diff
@@ -18,6 +18,14 @@
   display: none !important;
 }
 
+.h-auto {
+  height: auto;
```
Comment (MeghalBisht) [inline app/styles/partials/utils.scss:22]:
```suggestion
  height: auto !important;
```

File: app/styles/partials/utils.scss [hunk 6]
```diff
@@ -18,6 +18,14 @@
...
+}
+
+.h-100 {
+  height: 100%;
```
Comment (MeghalBisht) [inline app/styles/partials/utils.scss:26]:
```suggestion
  height: 100% !important;
```

**Comment by sans2801** [comment]
image
@MeghalBisht ma'am actually the _.event.card_ class (having _height: auto_) gets precedence over the _h-100_ class here, that's why the _height: 100%_ property gets neglected.
Thay's why I had used _.height.h-auto_ and _.height.h-100_ here so that they would get priority

**Comment by MeghalBisht** [comment]
> Thay's why I had used _.height.h-auto_ and _.height.h-100_ here so that they would get priority

Yes, now !important does the same work

**Comment by sans2801** [comment]
@MeghalBisht ma'am is the number of commits too large? Should I squash them?

**Comment by MeghalBisht** [comment]
No

**Review by sachinchauhan2889 — CHANGES_REQUESTED** [review]
desired result:-


Current result:-


You can achieve this by manipulating height percent of div(containing share icon and tags) and its above anchor tag.

**Comment by sans2801** [comment]
@sachinchauhan2889 yes sir, working on it

**Comment by MeghalBisht** [comment]
@sans2801 any update?

**Comment by sans2801** [comment]
@MeghalBisht yes ma'am, apologies for the delay
Can I get time till tomorrow evening.

**Comment by sans2801** [comment]
@MeghalBisht , ma'am I've implemented the suggestions, please let me know in case of any shortcomings

File: app/templates/components/event-card.hbs [hunk 7]
```diff
@@ -9,25 +9,42 @@
...
+              <SmartOverflow @class="header">
+                {{this.event.name}}
+              </SmartOverflow>
+              <div class="meta mb-2">
+                <span class="date">
+                  <Public::EventDateTime @event={{@event}} @noBreak={{true}} />
+                </span>
+              </div>
+              <Public::EventVenue @event={{@event}} />
+            </a>
```
Comment (MeghalBisht) [inline app/templates/components/event-card.hbs:45]: duplication

File: app/styles/partials/utils.scss [hunk 8]
```diff
@@ -18,6 +18,18 @@
   display: none !important;
 }
 
+.h-auto {
+  height: auto !important;
+}
+
+.h-100 {
+  height: 100% !important;
+}
```
Comment (MeghalBisht) [inline app/styles/partials/utils.scss:27]: use h-full class

File: app/styles/partials/utils.scss [hunk 9]
```diff
@@ -18,6 +18,18 @@
...
+
+.h-80perc {
+  height: 80% !important;
+}
+
```
Comment (MeghalBisht) [inline app/styles/partials/utils.scss:32]: why 80?

File: app/templates/components/event-card.hbs (hunk 7 above)
Comment (sans2801) [reply app/templates/components/event-card.hbs:45]: Yes ma'am, I agree there's repetition with very little change in the anchor tag code. What I meant to achieve was to make the height of the 'main content' anchor tag to 80% of the parent event card, so the hashtags would be pushed down. 
I am unable to implement a single inline conditional statement for the same.
I can implement a helper function to dealt with the conditions and reduce it to an inline function.
Would like to receive your suggestions on the same.

File: app/styles/partials/utils.scss (hunk 8 above)
Comment (sans2801) [reply app/styles/partials/utils.scss:27]: yes ma'am, understood

File: app/templates/components/event-card.hbs (hunk 7 above)
Comment (MeghalBisht) [reply app/templates/components/event-card.hbs:45]: Use helpers

File: app/styles/partials/utils.scss (hunk 9 above)
Comment (sans2801) [reply app/styles/partials/utils.scss:32]: Initially
image

Now
image

80% height seemed to give the desired output

**Comment by sans2801** [comment]
@MeghalBisht ma'am I have dealt with the repetitive code

**Comment by sans2801** [comment]
image
did the changes

**Review by MeghalBisht — APPROVED** [review]

**Comment by sans2801** [comment]
@MeghalBisht Hello ma'am, are there any concerns regarding this PR

**Review by MeghalBisht — APPROVED** [review]

**Pull request merged**